File: packages/libbird/android/src/main/java/dev/flutterbird/ladybird/WebContentService.java
package dev.flutterbird.ladybird;

import android.content.Context;
import android.content.Intent;
import android.os.Message;
import android.util.Log;

public final class WebContentService extends LadybirdServiceBase {
    public WebContentService() {
        super("WebContentService");
        nativeInit();
    }

    @Override
    boolean handleServiceSpecificMessage(Message msg) {
        return false;
    }

    private void bindRequestServer(int ipcFd) {
        LadybirdServiceConnection connector = new LadybirdServiceConnection(ipcFd, resourceDir);
        connector.onDisconnect = () -> Log.e(tag, "RequestServer disconnected");
        bindService(
                new Intent(this, RequestServerService.class),
                connector,
                Context.BIND_AUTO_CREATE);
    }

    private void bindImageDecoder(int ipcFd) {
        LadybirdServiceConnection connector = new LadybirdServiceConnection(ipcFd, resourceDir);
        connector.onDisconnect = () -> Log.e(tag, "ImageDecoder disconnected");
        bindService(
                new Intent(this, ImageDecoderService.class),
                connector,
                Context.BIND_AUTO_CREATE);
    }

    private void bindCompositor(int ipcFd) {
        LadybirdServiceConnection connector = new LadybirdServiceConnection(ipcFd, resourceDir);
        connector.onDisconnect = () -> Log.e(tag, "Compositor disconnected");
        bindService(
                new Intent(this, CompositorService.class),
                connector,
                Context.BIND_AUTO_CREATE);
    }

    private native void nativeInit();

    static {
        System.loadLibrary("webcontentservice");
    }
}

File: packages/libbird/android/src/main/kotlin/org/serenityos/ladybird/WebContentService.kt
package org.serenityos.ladybird

import android.content.Context
import android.content.Intent
import android.os.Message
import android.util.Log

class WebContentService : LadybirdServiceBase("WebContentService") {
  override fun handleServiceSpecificMessage(msg: Message): Boolean = false

  init {
    nativeInit()
  }

  private fun bindRequestServer(ipcFd: Int) {
    val connector = LadybirdServiceConnection(ipcFd, resourceDir)
    connector.onDisconnect = { Log.e("WebContentService", "RequestServer died") }
    bindService(Intent(this, RequestServerService::class.java), connector, Context.BIND_AUTO_CREATE)
  }

  private fun bindImageDecoder(ipcFd: Int) {
    val connector = LadybirdServiceConnection(ipcFd, resourceDir)
    connector.onDisconnect = { Log.e("WebContentService", "ImageDecoder died") }
    bindService(Intent(this, ImageDecoderService::class.java), connector, Context.BIND_AUTO_CREATE)
  }

  external fun nativeInit()

  companion object {
    init {
      System.loadLibrary("webcontentservice")
    }
  }
}
